tests/provision/reboot/test.sh

```diff
@@ -70,38 +152,70 @@ rlJournalStart
+            rlAssertGrep "reboot: Reboot finished" $rlRun_LOG
+
+            rlRun -s "tmt -vv run                --scratch -i $run_connect $provision                                reboot --step provision --command reboot" \
+                0 "Soft reboot with custom command"
+            rlAssertGrep "reboot: Rebooting guest using soft mode." $rlRun_LOG
+            rlAssertGrep "reboot: Reboot finished" $rlRun_LOG
+            rlAssertGrep "cmd: reboot" $rlRun_LOG
+
+            rlRun -s "tmt -vv run                --scratch -i $run_connect $provision --soft-reboot '$custom_reboot' reboot --step provision" \
+                2 "Soft reboot with custom soft-reboot command"
+            rlAssertGrep "reboot: Rebooting guest using soft mode." $rlRun_LOG
             rlAssertGrep "Custom soft and hard reboot commands are allowed only with the '--feeling-safe' option." $rlRun_LOG
 
-            rlRun -s "tmt -vv --feeling-safe run --scratch -i $run_connect $provision --soft-reboot '$custom_reboot' reboot --step provision"
-            rlAssertGrep "Reboot finished" $rlRun_LOG
+            rlRun -s "tmt -vv --feeling-safe run --scratch -i $run_connect $provision --soft-reboot '$custom_reboot' reboot --step provision" \
+                0 "Soft reboot with custom soft-reboot command and --feeling-safe"
+            rlAssertGrep "reboot: Rebooting guest using soft mode." $rlRun_LOG
+            rlAssertGrep "reboot: Reboot finished" $rlRun_LOG
             rlAssertGrep "cmd: /bin/bash -c '$custom_reboot'" $rlRun_LOG
 
-            # Hard reboot
-            rlRun -s "tmt -vv run                --scratch -i $run_connect $provision --hard-reboot '$custom_reboot' reboot --step provision" 2
+            # - Systemd soft-reboot ---------------------------------------------------------------
+            rlRun -s "tmt -vv run --scratch -i $run_connect $provision reboot --systemd-soft --step provision" \
+                0 "Systemd soft-reboot"
+            rlAssertGrep "reboot: Rebooting guest using systemd-soft mode." $rlRun_LOG
+            rlAssertGrep "reboot: Reboot finished" $rlRun_LOG
+            rlAssertGrep "cmd: systemctl soft-reboot" "$run_connect/log.txt"
+
+            rlRun -s "tmt -vv run --scratch -i $run_connect $provision reboot --systemd-soft --command 'systemctl soft-reboot' --step provision" \
+                0 "Systemd soft-reboot with custom command"
+            rlAssertGrep "reboot: Rebooting guest using systemd-soft mode." $rlRun_LOG
+            rlAssertGrep "reboot: Reboot finished" $rlRun_LOG
+            rlAssertGrep "cmd: systemctl soft-reboot" "$run_connect/log.txt"
+
+            rlRun -s "tmt -vvv run --scratch -i $run_connect $provision reboot --systemd-soft --command 'dummy' --step provision" \
+                2 "systemd soft-reboot with invalid custom command"
+            rlAssertGrep "reboot: Rebooting guest using systemd-soft mode." $rlRun_LOG
+            rlAssertGrep "dummy: command not found" $rlRun_LOG
```

Comment:
Shouldn't this also be disallowed according to "Custom soft and hard reboot commands are allowed only with the '--feeling-safe' option."?

Reply:
Well the documentation of that nuance is missing at least in some places :)

Reply:
Which places? I can fix it, just need to know where it's missing. `provision/connect` documents `--{soft,hard}-reboot` commands, `reboot` documents `--command`.